firmware/user/display/oled.c

```diff
@@ -83,42 +271,98 @@ typedef enum
+     *                     Valid command sequence 1: 26h ;2Fh.
+     *                     Valid command sequence 2: 27h ;2Fh.
+     *                     Valid command sequence 3: 29h ;2Fh.
+     *                     Valid command sequence 4: 2Ah ;2Fh.
+     *           false - Stop scrolling that is configured by command 26h/27h/29h/2Ah.
+     *
+     *   0 PARAMETERS: on ? SSD1306_ACTIVATE_SCROLL : SSD1306_DEACTIVATE_SCROLL
+     */
     SSD1306_DEACTIVATE_SCROLL = 0x2E,
     SSD1306_ACTIVATE_SCROLL = 0x2F,
+
+    /**
+     *
+     * @param A - Set No. of rows in top fixed area. The No. of rows in top fixed area
+     *            is referenced to the top of the GDDRAM (i.e. row 0).[RESET = 0]
+     *
+     * @param B - Set No. of rows in scroll area. This is the number of rows to be used
+     *            for vertical scrolling. The scroll area starts in the first row below
+     *            the top fixed area. [RESET = 64]
+     *
+     *  2 PARAMETERS: a and b.
+     */
     SSD1306_SET_VERTICAL_SCROLL_AREA = 0xA3,
+
+    /**
+     *
+     * @ param A - Set vertical shift by COM from 0d~63d The value is reset to 00h after RESET
+     *
+     *  1 PARAMETER: a
+	 */
+	SSD1306_SET_DISPLAY_OFFSET = 0xD3,
+
+    //Special commands added for our interpreter
+    PCD_CMD0 = 0xF0,
+    PCD_CMD1 = 0xF1,
+    PCD_CMD2 = 0xF2,
+    PCD_CMD3 = 0xF3,
+    PCD_COND0 = 0xF4,
+    PCD_COND1 = 0xF5,
+    PCD_COND2 = 0xF6,
+    PCD_COND3 = 0xF7,
+    PCD_END  = 0xFF,
 } SSD1306_cmd;
 
+
 //==============================================================================
-// Internal Function Declarations
+// Script of commands which get executed at system start
 //==============================================================================
 
-void ICACHE_FLASH_ATTR setContrastControl(uint8_t contrast);
-void ICACHE_FLASH_ATTR entireDisplayOn(bool ignoreRAM);
-void ICACHE_FLASH_ATTR setInverseDisplay(bool inverse);
-void ICACHE_FLASH_ATTR setDisplayOn(bool on);
-
-void ICACHE_FLASH_ATTR activateScroll(bool on);
-
-void ICACHE_FLASH_ATTR setMemoryAddressingMode(memoryAddressingMode mode);
-void ICACHE_FLASH_ATTR setColumnAddress(uint8_t startAddr, uint8_t endAddr);
-void ICACHE_FLASH_ATTR setPageAddress(uint8_t startAddr, uint8_t endAddr);
-
-void ICACHE_FLASH_ATTR setDisplayStartLine(uint8_t startLineRegister);
-void ICACHE_FLASH_ATTR setSegmentRemap(bool colAddr);
-void ICACHE_FLASH_ATTR setMultiplexRatio(uint8_t ratio);
-void ICACHE_FLASH_ATTR setComOutputScanDirection(bool increment);
-void ICACHE_FLASH_ATTR setDisplayOffset(uint8_t offset);
-void ICACHE_FLASH_ATTR setComPinsHardwareConfig(bool sequential, bool remap);
-
-void ICACHE_FLASH_ATTR setDisplayClockDivideRatio(uint8_t divideRatio, uint8_t oscFreq);
-void ICACHE_FLASH_ATTR setPrechargePeriod(uint8_t phase1period, uint8_t phase2period);
-void ICACHE_FLASH_ATTR setVcomhDeselectLevel(VcomhDeselectLevel level);
+static const uint8_t displayInitStartCommands[] RODATA_ATTR =
```

Comment:
You don't _have_ to put this in ROM with `RODATA_ATTR`. It's only 51 bytes; we can spare that in RAM. This is also used independent of swadge mode, which makes RAM more attractive.

`IsoSphereVertices[]` and `IsoSphereIndices[]` make more sense in ROM, since they're 480 + 252 bytes, and only used in one mode.

Reply:
I will do that as well, but, the array is only accessed a tiny bit at a time, so no reason not to go in RAM.
Also, relax,  IsoSphere is already in ROM.

Reply:
I'm relaxed 🙂. I mentioned IsoSphere because I noticed you already put it in ROM is all